src/lib/sys.c

```diff
@@ -59,6 +60,12 @@ int run_command_full_params(const char *stdout_file, const char *stderr_file, ch
 	int pid, ret, child_ret;
 	const char *cmd = params[0];
 	int null_fd;
+	int pipe_fds[2];
+
+	if (pipe(pipe_fds) < 0) {
```

Comment:
Use pipe2(pipe_fds, O_CLOEXEC) to prevent leaking the pipe

Reply:
okay

Reply:
Looks like when I build with it, it is just not able to find its declaration

Reply:
65 |  if (pipe2(pipe_fds, O_CLOEXEC) < 0) {
      |      ^~~~~
      |      pipe
src/lib/sys.c:65:6: warning: nested extern declaration of ‘pipe2’ [-Wnested-externs]
  CCLD     swupd

Reply:
According to the pipe2 manpage you need to

       #define _GNU_SOURCE
       #include <fcntl.h>
       #include <unistd.h>

Adding _GNU_SOURCE should fix this

Reply:
Yes it fixed it..

Reply:
Thanks.. looks like we support both POSIX and GNU linux extensions